File: java/prototype/exercise/GUIButton.java
package prototype.exercise;

public class GUIButton implements CloneableButton {

    private String color;
    private int xPosition;
    private int yPosition;
    private String content;

    public GUIButton(String color, int xPosition, int yPosition, String content) {
        this.color = color;
        this.xPosition = xPosition;
        this.yPosition = yPosition;
        this.content = content;
    }

    public GUIButton cloneButton() {
        return new GUIButton(this.color, this.xPosition, this.yPosition, this.content);
    }

}

File: C#/Prototype/Excercise/GUIButton.cs
namespace Prototype.Excercise;

public sealed class GUIButton : IClonableButton
{
    private readonly string _color;
    private readonly int _xPosition;
    private readonly int _yPosition;
    private readonly string _content;

    public GUIButton(string color, int xPosition, int yPosition, string content)
    {
        _color = color;
        _xPosition = xPosition;
        _yPosition = yPosition;
        _content = content;
    }

    public GUIButton CloneButton()
        => new(_color, _xPosition, _yPosition, _content);
}
